flang/lib/Lower/ConvertCall.cpp

```diff
@@ -942,6 +964,18 @@ static PreparedDummyArgument preparePresentUserCallActualArgument(
       (passingPolymorphicToNonPolymorphic ||
        !Fortran::evaluate::IsSimplyContiguous(expr, foldingContext));
 
+  // Create dummy type with actual argument rank when the dummy is an assumed
+  // rank. That way, all the operation to create dummy descriptors are ranked if
+  // the dummy is ranked, which allows simple code generation.
```

Comment:
I am not sure I am following the comment.  "if the dummy is ranked" seems to contradict the previous condition that "dummy is an assumed rank".  Can you please clarify?

Reply:
Indeed, thanks for reading it! updated to "to create dummy descriptors are ranked if the actual argument is ranked"